src/native/fs/fs_napi.cpp

```diff
@@ -1939,6 +1943,52 @@ struct DirOpen : public FSWorker
     }
 };
 
+struct TellDir : public FSWrapWorker<DirWrap>
+{
+    std::string _dir_path;
+    long int _tell_res;
+    TellDir(const Napi::CallbackInfo& info)
+        : FSWrapWorker<DirWrap>(info)
+    {
+        Begin(XSTR() << "TellDir " << DVAL(_wrap->_path));
+    }
+    virtual void Work()
+    {
+        DIR* dir = _wrap->_dir;
+        std::string _dir_path = _wrap->_path;
+        DBG1("FS::Telldir::Work: " << DVAL(_dir_path));
+        _tell_res = telldir(dir);
+        if (_tell_res == -1) SetSyscallError();
+    }
+    virtual void OnOK()
+    {
+        DBG0("FS::Telldir::OnOK: " << DVAL(_dir_path) << DVAL(_tell_res));
+        Napi::Env env = Env();
+        auto res = Napi::Object::New(env);
+        res["telldir_res"] = Napi::Number::New(env, _tell_res);
+        _deferred.Resolve(res);
+    }
+};
+
+struct SeekDir : public FSWrapWorker<DirWrap>
+{
+    std::string _dir_path;
+    long int _seek_pos;
```

Comment:
why long int?
I see it's a long - https://man7.org/linux/man-pages/man3/seekdir.3.html

Reply:
int is optional in these types - long=long int , short=short int, signed=signed int, unsigned=unsigned int.